File: past-competitions/2018/src/brett-cpp/Prob04.c
#include <string.h>
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

int main(int argc, char *argv[])
{
	// open file for reading
	FILE *inputFile = fopen("Prob04.in.txt", "r");
	int testCases;

	// read number of test cases
	fscanf(inputFile, "%d\n", &testCases);
	
	int rockCount, paperCount, scissorsCount;

	// execute test cases
	while(testCases > 0){
		// read first character
		char nextChar = fgetc(inputFile);
		// reset counts
		rockCount = 0;
		paperCount = 0;
		scissorsCount = 0;

		// read each character until we hit a newline
		while(nextChar != '\n'){
			// count rock/paper/scissors (ignore spaces)
			if(nextChar == 'R'){
				rockCount++;
			}
			else if(nextChar == 'P'){
				paperCount++;
			}
			else if(nextChar == 'S'){
				scissorsCount++;
			}

			// read next character
			nextChar = fgetc(inputFile);
		}

		if(rockCount == 1 && paperCount == 0){
			// rock wins if there's one rock and no papers to beat it
			printf("ROCK\n");
		}
		else if(paperCount == 1 && scissorsCount == 0){
			// paper wins if there's one paper and no scissors to beat it
			printf("PAPER\n");
		}
		else if(scissorsCount == 1 && rockCount == 0){
			// scissors win if there's one scissors (scissor?) and no rocks to beat it
			printf("SCISSORS\n");
		}
		else{
			// either there's a multiple of something, or all three results are represented
			// either way, it's a tie
			printf("NO WINNER\n");
		}

		testCases = testCases - 1;
	}

	// close the file
	fclose(inputFile);
}

File: past-competitions/2018/src/brett-python/Prob04.py
# Open the input file
with open("Prob04.in.txt", "rt") as inputFile:
    # Read the number of test cases (trim out the newline)
    cases = int(inputFile.readline().replace("\n", ""))
    # For each test case
    for caseNum in range(cases):
        # Read the line
        line = inputFile.readline().replace("\n", "")
        # Split the string on spaces
        responses = line.split(" ")
        # Start counters
        rock = 0
        paper = 0
        scissors = 0
		# Count responses
        for response in responses:
            if response == 'R':
                rock = rock + 1
            elif response == 'P':
                paper = paper + 1
            else:
                scissors = scissors + 1
		# Determine the winner - must be alone, 
		# and whatever beats it must be 0
        if rock == 1 and paper == 0:
            print("ROCK")
        elif paper == 1 and scissors == 0:
            print("PAPER")
        elif scissors == 1 and rock == 0:
            print("SCISSORS")
        else:
            print("NO WINNER")

        
